_posts/2022/2022-10-31-use-zero-for-integer-float-comparison.md

```diff
@@ -0,0 +1,119 @@
+---
+title: "Use zero? for comparison of numerics like Integer, Float and BigDecimal"
+description: "Other languages don't have this"
+layout: article
+category: ruby
+image:
+  base: "2022/use-zero-for-integer-float-comparison"
+  alt: "A zero on a sign on the wall"
+  credit: "Bernard Hermant"
+  source: "https://unsplash.com/photos/rfpSOlH1JlQ"
+---
+
+Compared to other similar languages Ruby often prioritises readability (and joy) when it come to its syntax and methods in its standard library.
+
+An example of this is some syntactic sugar for comparing a value to zero.
+
+See the documentation of the `.zero?` method on [`Integer`](https://ruby-doc.org/core-3.1.2/Integer.html#zero-3F-method), [`Float`](https://ruby-doc.org/core-3.1.2/Float.html#zero-3F-method), [`Numeric`](https://ruby-doc.org/core-3.1.2/Numeric.html#zero-3F-method) and [`BigDecimal`](https://ruby-doc.org/stdlib-3.1.2/libdoc/bigdecimal/rdoc/BigDecimal.html#zero-3F-method).
+
+
+## Instead of...
+
+...checking whether a value is equal to zero in a conditional:
+
+```ruby
+if number == 0
+  :yes
+else
+  :no
+end
+```
+
+
+## Use...
+
+...the built-in `#zero?` method on all numeric types:
+
+```ruby
+if number.zero?
+  :yes
+else
+  :no
+end
+```
+
+
+## Why?
+
+The Ruby-ish syntax in this case is clearer and less error prone.
+
+It is very easy to type `number = 0` and thus assign a variable in the conditional rather than compare (although you might use tests to protect yourself from this particular error).
```

Comment:
```suggestion
With the first approach/in the first example, it is very easy to mistakenly type `number = 0` and thus assign a variable in the conditional rather than perform a comparison (although you might use tests to protect yourself from that particular error).
```